packages/cashier-v2/src/lib/PaymentAgent/lib/PaymentAgentWithdrawal/provider/PaymentAgentWithdrawalProvider.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { createContext, useCallback, useContext, useState } from 'react';
+import { useActiveAccount, usePaymentAgentWithdrawal } from '@deriv/api-v2';
+import type { THooks } from '../../../../../hooks/types';
+import type { TCurrency } from '../../../../../types';
+import type { TConfirm, TPaymentAgentWithdrawalModuleProps, TReceipt } from '../types';
+import {
+    getPaymentAgentListedWithdrawalValidationSchema,
+    getPaymentAgentUnlistedWithdrawalValidationSchema,
+} from '../utils';
+
+type TRequestTryPaymentAgentWithdrawalParams = Omit<
+    Parameters<ReturnType<typeof usePaymentAgentWithdrawal>['mutateAsync']>[0]['payload'],
+    'currency' | 'verification_code'
+>;
+
+type TRequestPaymentAgentWithdrawalParams = TRequestTryPaymentAgentWithdrawalParams & {
+    paymentAgentEmail: TReceipt['paymentAgentEmail'];
+    paymentAgentPhoneNumbers: TReceipt['paymentAgentPhoneNumbers'];
+    paymentAgentUrls: TReceipt['paymentAgentUrls'];
+};
+
+type TGetPaymentAgentWithdrawalValidationSchema = (
+    maxWithdrawal?: THooks.PaymentAgentList[number]['max_withdrawal'],
+    minWithdrawal?: THooks.PaymentAgentList[number]['min_withdrawal']
+) => ReturnType<typeof getPaymentAgentListedWithdrawalValidationSchema>;
+
+type TPaymentAgentWithdrawalContext = {
+    activeAccount?: THooks.ActiveAccount;
+    getPaymentAgentWithdrawalValidationSchema: TGetPaymentAgentWithdrawalValidationSchema;
+    isLoading: boolean;
+    isTryWithdrawalSuccessful: boolean;
+    isUnlistedWithdrawal: boolean;
+    isWithdrawalRequestSubmitting: boolean;
+    isWithdrawalSuccessful: boolean;
```

Comment:
```suggestion
    isUnlistedWithdrawal: boolean;
    isWithdrawalRequestSubmitting: boolean;
    isWithdrawalSuccessful: boolean;
```
Since in these 3 states, for all time only one of these can happen. I suggest using a string union as type for this.

eg;
const withdrawalStatus = "unlisted" | "request_submitting" | "successful";


multiple boolean is actually hard to maintain. if they belong to the same group, best to use string union. if u need it to be precalculated as boolean in the consumer, u still can do this when passing down the context data:

```
isUnlistedWithdrawal: withdrawalStatus === "unlisted"
isWithdrawalRequestSubmitting: withdrawalStatus === "request_submitting" 
```